bpfd/src/multiprog/tc.rs

```diff
@@ -195,10 +195,11 @@ impl TcDispatcher {
                 let path = format!("{base}/dispatcher_{if_index}_{}/link_{k}", self.revision);
                 new_link.pin(path).map_err(BpfdError::UnableToPinLink)?;
             } else {
-                let program_bytes = if v.data.path.clone().contains(BYTECODE_IMAGE_CONTENT_STORE) {
-                    get_bytecode_from_image_store(v.data.path.clone()).await?
+                let path = v.data.location.path().await?.to_string_lossy().to_string();
+                let program_bytes = if path.contains(BYTECODE_IMAGE_CONTENT_STORE) {
```

Comment:
Now this will just be something like let program_bytes = v.data.location.get_program_bytes(&program.data().section_name)?